spec/21-http_response_header_format.md

```diff
@@ -107,6 +96,15 @@ There are two additional options that tracing systems MAY follow:
 - A component that makes a deferred or delayed recording decision may communicate the priority of a recording by setting `sampled` flag to `1` for a subset of requests.
 - A component may also fall back to probability sampling and set the `sampled` flag to `1` for the subset of requests.
 
+##### Random Trace ID Flag
+
+The second least significant bit of the trace-flags field denotes the random-trace-id flag.
+
+The format and requirements for this are the same as those of the random-trace-id flag in the trace-flags field in the `traceparent` request header.
+
+For details, see the trace-flags section under [traceparent Header Field Values](#traceparent-Header-Field-Values).
```

Comment:
_issue:_

Thinking about this again, one more aspect came to my mind.

I think it is pretty clear how participants are ought to behave that _start_ a trace. If they generate the trace id randomly, they set the flag in the response header to signal the randomness upstream. 

But what about participants which continue a trace that has been started upstream? They receive a value for the flag in the request header from the upstream participant. They also potentially receive multiple response headers from requests they send downstream. What if the flag values are not identical? This should not happen in practice I think, but we still should spell out explicit rules here IMO.

It might be fairly obvious I think: When continuing a trace we always need to respect the decision of the participant that generated the trace id. Only that participant knows how it generated the id. Thus, a participant that continues the trace MUST set the flag to the same value as in the incoming request traceparent header. Even if it is obvious, I think we should still state this explicitly.

That said, the benefit of the flag in the response header is questionable anyway. You are telling the upstream participant something it already knows anyway. But of course we need it for consistency, since the flag has value in the response header when the participant started the trace.

Reply:
Thanks Bastian. Regarding "You are telling the upstream participant something it already knows anyway. ", there is a case where the upstream participant didn't generate the trace-id (i.e., the trace was started by the downstream participant), and in that case the upstream participant would benefit from this random-trace-id flag in the response header. I will add a note to this effect.

Apart from the above, given that you have already addressed how the flag should be set on trace continuation, do you see the need for clarifying anything else here?

Reply:
I was thinking about some explicit normative language like the following:

* A participant that starts or restarts the trace by generating a new `trace-id` MUST apply the rules from the section on the random trace ID for `traceparent` for the `traceresponse` header as well.
* A participant that uses the same `trace-id` in `traceresponse` as it received in the incoming `traceparent` header MUST also set the `random-trace-id` flag to the same value that was found in that `traceparent` header.
* If a participant did not receive an incoming `traceparent` header, but it received a `traceresponse` header from downstream, and it wants to communicate the `trace-id` from that `traceresponse` header upstream, it MUST use the same `random-trace-id` flag value as in that `traceresponse` header.

Side note: The last bullet point refers to the non-normative example in the section "Load Balancer Deferred Sampling".

Alternatively, we can also drop this point for now and revisit it later. That is merge the PR as-is, and iterate on the normative wording later on.

Reply:
Thanks Bastian. Filed #540 to track this suggestion.

Reply:
Thanks for moving that out into its own issue. 👍